test-common/test-common-api/src/main/java/org/apache/kafka/common/test/api/ClusterInstance.java

```diff
@@ -225,4 +235,56 @@ default void waitAcls(AclBindingFilter filter, Collection<AccessControlEntry> en
         }
     }
 
+    default <B extends KafkaBroker> void verifyTopicDeletion(String topic, int numPartitions) throws Exception {
+        verifyTopicDeletion(topic, numPartitions, brokers().values());
+    }
+
+    default <B extends KafkaBroker> void verifyTopicDeletion(String topic, int numPartitions, Collection<B> brokers) throws Exception {
+        List<TopicPartition> topicPartitions = IntStream.range(0, numPartitions)
+            .mapToObj(partition -> new TopicPartition(topic, partition))
+            .collect(Collectors.toList());
+
+        // Ensure that the topic-partition has been deleted from all brokers' replica managers
+        TestUtils.waitForCondition(() -> brokers.stream().allMatch(broker ->
```

Comment:
Since `clusterInstance` support shutdown broker, using `aliveBroker` is better solution.
You can refer https://github.com/apache/kafka/pull/17085